### ISSUE TITLE
[v2] docker compose up fails if directory name contains only digits

### ISSUE THREAD

**Issue by stephen-turner** [docker/compose#8691]
**Description**

Using docker compose v2.0.0-rc.3, I tried to run `docker compose up` in a directory where the name contains only digits. It failed because the generated container names were invalid because of an initial underscore:
```
sret1@User-PC:/mnt/c/Users/User/work/tmp/for-win/12018$ docker-compose up
[+] Running 0/1
 ⠙ Container _alpine_1  Creating                                                                   0.1s
Error response from daemon: Invalid container name (_alpine_1), only [a-zA-Z0-9][a-zA-Z0-9_.-] are allowed
```
It appears that leading digits are stripped from the directory name. When I changed the directory name to 12a18, I got this:
```
sret1@User-PC:/mnt/c/Users/User/work/tmp/for-win/12a18$ docker-compose up
[+] Running 1/1
 ⠿ Container a18_alpine_1  Created                                                                 1.2s
```
This doesn't happen with compose v1.29.2:
```
sret1@User-PC:/mnt/c/Users/User/work/tmp/for-win/12018$ docker-compose up
Creating 12018_alpine_1 ... done
```
Of course, it's also fine if I use the `-p` argument to override the directory name:
```
sret1@User-PC:/mnt/c/Users/User/work/tmp/for-win/12018$ docker-compose -pfoo up
[+] Running 1/1
 ⠿ Container foo_alpine_1  Created                                                                 1.2s
```

**Output of `docker compose version`:**

```
docker-compose version
Docker Compose version v2.0.0-rc.3
```

**Output of `docker info`:**

```
$ docker info
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Build with BuildKit (Docker Inc., v0.6.3)
  compose: Docker Compose (Docker Inc., v2.0.0-rc.3)
  scan: Docker Scan (Docker Inc., v0.8.0)

Server:
 Containers: 1
  Running: 0
  Paused: 0
  Stopped: 1
 Images: 9
 Server Version: 20.10.8
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: e25210fe30a0a703442421b0f60afac609f950a3
 runc version: v1.0.1-0-g4144b63
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.10.16.3-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 3.839GiB
 Name: docker-desktop
 ID: SHC3:3NUN:VXK3:PD2R:FDKR:WZBZ:6VEQ:ORLT:FEGP:HEE7:YLRP:V5KP
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Additional environment details:**

Docker Desktop in WSL 2

**Comment by ulyssessouza** [comment]
I've just tried with the `v2` branch and that works for me.
Looks like it's a duplicate of docker/for-mac#5974 should be already fixed by https://github.com/compose-spec/compose-go/pull/189.

**Comment by mat007** [comment]
@stephen-turner do you want to check with Desktop 4.0.1, or should we just close this issue?

**Comment by stephen-turner** [comment]
Confirmed fixed in Desktop 4.1.0 (compose v2.0.0). Thank you.
